### ISSUE TITLE
defineConstant not working

### ISSUE THREAD

**Issue by kidsang** [microsoft/pyright#3717]
**Describe the bug**
Configuration setting "defineConstant" not working. 
Pyright's reachability analysis dosent seen to be affect by it.
![image](https://user-images.githubusercontent.com/1716762/179535086-fb6d1993-cc65-4ce0-a6b7-6c9bc898e6bc.png)

**To Reproduce**
1. Add following setting in pyrightconfig.json
```json
"defineConstant": {
	"DEBUG": false,
}
```
2. Add following code in projcect script file
```python
DEBUG = False
if DEBUG:
	a = 1
	print(a)
```
3. Pyright still think statments under "if DEBUG:" is rechable (color's not changed).

**Expected behavior**
Statements under "if Debug:" become unreachable.

**Comment by kidsang** [comment]
Version of Pylance language server is 2022.7.30
![image](https://user-images.githubusercontent.com/1716762/179538224-87c525dd-8406-4034-ad69-16e9e1a21571.png)

**Comment by erictraut** [comment]
I'm able to repro the problem you're seeing with pylance. It didn't show up in any of our internal testing or with our unit tests because it works fine in in debug mode. It appears only in production mode, where we use background threads for processing. This will be fixed in the next release.

**Comment by kidsang** [comment]
> I'm able to repro the problem you're seeing with pylance. It didn't show up in any of our internal testing or with our unit tests because it works fine in in debug mode. It appears only in production mode, where we use background threads for processing. This will be fixed in the next release.

Thx for the reply, cant wait to see it fixed!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.262, which I just published. It will also be included in a future version of pylance.
